src/components/NavBar.js

```diff
@@ -1,37 +1,58 @@
-import React from 'react';
+import React, {useContext} from 'react';
 import logo from '../assets/banana-01.png';
 import { useHistory, Link } from 'react-router-dom';
+import {AuthContext} from "../context/AuthContext";
 
 function NavBar() {
-  const history = useHistory();
+    const history = useHistory();
 
-  return (
-    <nav>
-        <Link to="/">
+    const {isAuth, isSignedOutFunction} = useContext(AuthContext);
+
+
+    return (
+        <nav>
+            <Link to="/">
           <span className="logo-container">
             <img src={logo} alt="logo"/>
             <h3>
               Banana Security
             </h3>
           </span>
-        </Link>
-
-      <div>
-        <button
-          type="button"
-          onClick={() => history.push('/signin')}
-        >
-          Log in
-        </button>
-        <button
-          type="button"
-          onClick={() => history.push('/signup')}
-        >
-          Registreren
-        </button>
-      </div>
-    </nav>
-  );
+            </Link>
+
+            <div>
+                {!isAuth ?
+                    <>
+                        <button
+                            type="button"
+                            onClick={() => history.push('/signin')}
+                        >
+                            Log in
+                        </button>
+                        <button
+                            type="button"
+                            onClick={() => history.push('/signup')}
+                        >
+                            Registreren
+                        </button>
+                    </>
+                    :
+                    <>
+                        <button
+                            type="button"
+                            onClick={isSignedOutFunction}
+                        >
+                            Sign out
+                        </button>
+                    </>
```

Comment:
De fragments zijn niet nodig, omdat het maar één child bevat
```suggestion
                    <button
                        type="button"
                        onClick={isSignedOutFunction}
                    >
                        Sign out
                    </button>
```